### DISCUSSION TITLE
Multi-n-graph best approach

### DISCUSSION THREAD

**Discussion by piyush-J** [pyg-team/pytorch_geometric#4585] [Q&A]
Hi, Thank you for the great documentation! I am a student I want to implement an n-graph data object for my project. This would be an extension of the [pair-data](https://pytorch-geometric.readthedocs.io/en/latest/notes/batching.html#pairs-of-graphs), where instead of 2 graphs, I want to create n graphs (n will also be an argument). 

What's the most pythonic way to do this (especially for the `__inc__` part)? Let's say the edge index will always have the naming convention (edge_index_1, edge_index_2, ..., edge_index_n) and similarly for x (x_1, x_2, ..., x_n). 

```
from torch_geometric.data import Data

class NGraphData(Data):
    def __init__(self, n, **kwargs):
        super().__init__()
        ....

    def __inc__(self, key, value, *args, **kwargs):
        if ....
        else:
            return super().__inc__(key, value, *args, **kwargs)
```

**Comment by Padarn** [comment]
maybe simply 
```
def __inc__(self, key, value, *args, **kwargs):
    if 'edge_index' in key:
        n = key.split("_")[-1]
        return getattr(self, f"x_{n}").size(0)
```
Could also use a regex if you feel its better

**Reply by piyush-J** [reply]
Thank you for your prompt reply. Will give it a try!

**Reply by rusty1s** [reply]
Batching of lists holding tensors should also be supported.
```python
data.edge_index_list = [edge_index1, edge_index2, edge_index3]
```
although it does not support customizing `__inc__` for each individual index currently.
For this, the approach from @Padarn might be a better alternative.
